testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/client/proxy/UriUsingIPv6Test.java

```diff
@@ -0,0 +1,108 @@
+package org.jboss.resteasy.test.client.proxy;
+
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import java.net.SocketException;
+import java.net.URISyntaxException;
+import java.net.URI;
+
+
+/**
+ * @tpSubChapter Resteasy-client
+ * @tpChapter Client tests
+ * @tpSince RESTEasy 4.0.0
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class UriUsingIPv6Test {
```

Comment:
Integration tests should be used when actual network communication is needed for the test, when a mix of client and server behavior is tested, when there's a user reproducer to be verified, etc.
In the case here, it might make sense to write an integration test, but that should really perform the invocation using the IPv6 address, if IPv6 is enabled. Judging from the test code, I think this test wouldn't pass if IPv6 was on, would it?